### ISSUE TITLE
salt from develop branch can't render pillar 'NoneType' object has no attribute 'endswith'

### ISSUE THREAD

**Issue by aboe76** [saltstack/salt#19702]
With the latest salt from develop branch I get the following error when running a highstate or pillar.item

```
salt 'salt.example.lan' state.highstate
salt.example.lan:
    Data failed to compile:
----------
    Pillar failed to render with the following messages:
----------
    Rendering Primary Top file failed, render error:
'NoneType' object has no attribute 'endswith'
```

Tested it against the 2014.7 no warnings.

Yesterday the salt development branch worked, something in the last 24 hours has produced this bug.

**Comment by cachedout** [comment]
Thanks for filing this. We're looking at this right now. The discussion is happening in #19571

**Comment by aboe76** [comment]
setting explicitly pillar_roots variable in salt master and minion config is a workaround for this.
e.g. uncomment the defaults in master and minion.

**Comment by cachedout** [comment]
Fixed by #19886
